zebra-chain/src/transaction/unmined/zip317.rs

```diff
@@ -139,3 +147,64 @@ fn conventional_actions(transaction: &Transaction) -> u32 {
 
     max(GRACE_ACTIONS, logical_actions)
 }
+
+/// Make ZIP-317 checks before inserting a transaction into the mempool.
+pub fn mempool_checks(
+    unpaid_actions: u32,
+    miner_fee: Amount<NonNegative>,
+    conventional_fee: Amount<NonNegative>,
+    transaction_size: usize,
+) -> Result<(), Error> {
+    // # Standard Rule
+    //
+    // > If a transaction has more than `block_unpaid_action_limit` "unpaid actions" as defined by the
+    // > Recommended algorithm for block template construction, it will never be mined by that algorithm.
+    // > Nodes MAY drop these transactions.
+    //
+    // <https://zips.z.cash/zip-0317#transaction-relaying>
+    if unpaid_actions > BLOCK_PRODUCTION_UNPAID_ACTION_LIMIT {
+        return Err(Error::UnpaidActions);
+    }
+
+    // # Standard Rule
+    //
+    // > Nodes that normally relay transactions are expected to do so for transactions that pay at least the
+    // > conventional fee as specified in this ZIP.
+    //
+    // <https://zips.z.cash/zip-0317#transaction-relaying>
+    if miner_fee < conventional_fee {
+        return Err(Error::FeeBelowConventional);
+    }
+
+    // # Standard rule
+    //
+    // > Minimum Fee Rate
+    // >
+    // > Transactions must pay a fee of at least 100 zatoshis per 1000 bytes of serialized size,
+    // > with a maximum fee of 1000 zatoshis. In zcashd this is `DEFAULT_MIN_RELAY_TX_FEE`.
+    //
+    // <https://github.com/ZcashFoundation/zebra/issues/5336#issuecomment-1506748801>
```

Comment:
Ah, I see where this first rule came from! This rule says that we can refuse to relay all of these transactions, but it is our choice which ones we relay. It doesn't force us to stop relaying any transactions.

In the ZIP editors meeting, we decided to be similar to `zcashd`, which is actually the next check you've already written.

So we can use these docs to justify the next check, but delete the first check, because it is too strict:
```suggestion
    // # Standard Rule
    //
    // > Nodes that normally relay transactions are expected to do so for transactions that pay at least the
    // > conventional fee as specified in this ZIP.
    //
    // <https://zips.z.cash/zip-0317#transaction-relaying>
    //
    // In Zebra, we use a similar minimum fee rate to `zcashd` v5.5.0 and later.
    // Transactions must pay a fee of at least 100 zatoshis per 1000 bytes of serialized size,
    // with a maximum fee of 1000 zatoshis. 
    //
    // <https://github.com/zcash/zcash/blob/9e856cfc5b81aa2607a16a23ff5584ea10014de6/src/amount.cpp#L15-L37>
    //
    // In zcashd this is `DEFAULT_MIN_RELAY_TX_FEE` and `LEGACY_DEFAULT_FEE`:
    // <https://github.com/zcash/zcash/blob/f512291ff20098291442e83713de89bcddc07546/src/main.h#L71-L72>
    // <https://github.com/zcash/zcash/blob/9e856cfc5b81aa2607a16a23ff5584ea10014de6/src/amount.h#L35-L36>
```